client/gateway/dappchain_gateway.go

```diff
@@ -93,6 +108,39 @@ func (tg *DAppChainGateway) AddContractMapping(from common.Address, to loom.Addr
 	return err
 }
 
+// AddTronContractMapping same as AddContractMapping but for Tron
+func (tg *DAppChainGateway) AddTronContractMapping(from common.Address, to loom.Address,
+	creator *client.Identity, contractTxHash string) error {
+	fromAddr, err := client.LoomAddressFromTronAddress(from)
+	if err != nil {
+		return err
+	}
+	txHash, err := hex.DecodeString(strings.TrimPrefix(contractTxHash, "0x"))
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Mapping contract %v to %v\n", fromAddr, to)
+
+	hash := ssha.SoliditySHA3(
+		ssha.Address(from),
+		ssha.Address(common.BytesToAddress(to.Local)),
+	)
+
+	sig, err := evmcompat.GenerateTypedSig(hash, creator.MainnetPrivKey, evmcompat.SignatureType_EIP712)
```

Comment:
Same sig type issue here that I've mentioned above

Reply:
fixed